### PR TITLE
feat: infer initial schema

### LINKED ISSUE #12: Tool to infer initial schema of dataset
**Is your feature request related to a problem? Please describe.**

To do any checks on the schemas of datasets we need to know the schema of the dataset that was initially loaded.

**Describe the solution you'd like**

A command line tool that takes a `csv` file as an input and produces the schema (see lars-reimann/safe-data-science#13) as output in a new file.

**Additional context (optional)**

* Best implemented in Kotlin, so our code serializer can be used.
* Possible library (**look for alternatives**): 
  * https://github.com/frictionlessdata/tableschema-java
  * https://github.com/jtablesaw/tablesaw

### PR DESCRIPTION
**Pull request by MShahzaib**
Closes #12.

### Summary of Changes

- added [Tablesaw](https://github.com/jtablesaw/tablesaw) for inferring the initial dataset schema
- implemented schema inference for the initial dataset schema with a Unit test

### CODE REVIEW

**Comment by MShahzaib** [comment]
@lars-reimann Please merge this after the [PR 79](https://github.com/lars-reimann/Safe-DS/pull/78)

File: DSL/com.larsreimann.safeds/src/main/kotlin/com/larsreimann/safeds/staticAnalysis/schema/SchemaInference.kt
Comment (lars-reimann) [inline DSL/com.larsreimann.safeds/src/main/kotlin/com/larsreimann/safeds/staticAnalysis/schema/SchemaInference.kt:41]: Is there a more specific exception type we could catch here?

File: DSL/com.larsreimann.safeds/src/test/resources/schema/testDataset.csv
Comment (lars-reimann) [inline DSL/com.larsreimann.safeds/src/test/resources/schema/testDataset.csv:1]: Where is this dataset from? How is it licensed? I guess it's the titanic dataset, right?

**Review by lars-reimann — CHANGES_REQUESTED** [review]
I'll have another look after #78 is merged.

File: DSL/com.larsreimann.safeds/src/test/resources/schema/testDataset.csv
Comment (MShahzaib) [reply DSL/com.larsreimann.safeds/src/test/resources/schema/testDataset.csv:1]: Yes, titanic dataset.

**Review by lars-reimann — APPROVED** [review]

**Pull request merged**
